inst/assumptions.m

```diff
@@ -111,10 +111,9 @@
       'd = x._assumptions.generator'
       'if d == {}:'
       '    astr = ""'
-      'elif all(d.values()):'  % all True so list them
-      '    astr = ", ".join(sorted([str(i) for i in d.keys()]))'
-      'else:'  % more complicated case, just the raw dict
-      '    astr = str(d)'
+      'else:'  % all True so list them
+      '    val, key = list(d.values()), list(d.keys())' % In Python3 d.values() don't support indexing
+      '    astr = ", ".join(sorted([(("" if val[i] else "~") + str(key[i])) for i in range(len(d.keys()))]))'
```

Comment:
Instead of indexing with `i` can this be something like `[... for (v, k) in zip(list(d.values(), list(d.keys()))]`?

or `[("" if d[k] else "~") + str(k) for k in d]`?

or `[("" if v else "~") + str(k) for (k,v) in d.items()]`?

(haven't tested which are portable between 2 and 3)